src/main/java/com/marklogic/kafka/connect/source/RowBatcherSourceTask.java

```diff
@@ -33,20 +51,79 @@ public String version() {
     @Override
     public final void start(Map<String, String> config) {
         logger.info("Starting");
-        Map<String, Object> parsedConfig = MarkLogicSourceConfig.CONFIG_DEF.parse(config);
-        logger.info("Started");
+        logger.info(String.valueOf(config));
+        this.parsedConfig = MarkLogicSourceConfig.CONFIG_DEF.parse(config);
+        DatabaseClientConfig databaseClientConfig = new DefaultDatabaseClientConfigBuilder().buildDatabaseClientConfig(parsedConfig);
+        this.databaseClient = new DefaultConfiguredDatabaseClientFactory().newDatabaseClient(databaseClientConfig);
+        dataMovementManager = databaseClient.newDataMovementManager();
     }
 
     @Override
     public List<SourceRecord> poll() throws InterruptedException {
         logger.info("poll not yet implemented");
 
+        rowBatcher =  getNewRowBatcher();
+        performPoll();
+
         Thread.sleep(1000);
         return null;
     }
 
+    protected RowBatcher<JsonNode> getNewRowBatcher() {
+        rowBatcher =  dataMovementManager.newRowBatcher(jsonHandle);
+        configureRowBatcher(parsedConfig, rowBatcher);
+        return rowBatcher;
+    }
+
+    protected void performPoll() {
+        dataMovementManager.startJob(rowBatcher);
+        rowBatcher.awaitCompletion();
+        rowBatcher = null;
```

Comment:
This should call `stopJob` after `awaitCompletion` (we don't stop the job for writing because we want to keep the job going forever essentially, but a read job stops once all the batches have been processed). 

I say that only because the docs recommend doing that for QueryBatcher - https://docs.marklogic.com/guide/java/data-movement#id_77234 . I think DMSDK's intent is that you should still call `stopJob` so that the underlying implementation can do any cleanup that it would like to do. I checked the tests in the Java Client and we do have lots of RowBatcher tests that only call `awaitCompletion` and not `stopJob` afterwards, but I think it's better practice to call `stopJob` as well.